src/main/java/command/FinderCommand.java

```diff
@@ -13,13 +13,13 @@ public FinderCommand(Printer printer) {
 
     protected CheatSheet getCheatSheetFromNameOrIndex() throws CommandException {
         CheatSheet desiredCheatSheet = null;
-        if (descriptionMap.containsKey(ArgumentFlagEnum.NAME)) {
+        if (descriptionMap.containsKey(ArgumentFlagEnum.NAME) && descriptionMap.get(ArgumentFlagEnum.NAME) != null) {

```

Comment:
Currently when I tried
`/delete /i 2`
The description map will return
```
NAME -> NULL
INDEX -> 2
```

So without the null check, it will always try to get the cheatSheet by name even when the user tries to get it by its index